app/services/strategy_service.rb

```diff
@@ -0,0 +1,9 @@
+class StrategyService
```

Comment:
I think this would fit well in the `lib/` directory. What about naming it something like `AuthenticationStrategies`?